### PR TITLE
Created deque/deque.cpp

### PR DESCRIPTION
**Pull request by jainaman1398**
**Fixes issue:**
Deque implementation using linked list is added

**Changes:**
deque.cpp is added in the deque folder

### CODE REVIEW

File: code/data_structures/src/queue/deque/deque.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
+#include <iostream>
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: You should have used templates for this.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            front->prev = tmp;
+            front = tmp;
+        }
+
+        count++;
+    }
+
+    int RemoveFront()
+    {
+        if ( isEmpty() ) {
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: For if and loops that have one line, you should omit the braces. This is applicable for all the file.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    void InsertFront(int element)
+    {
+        // Create a new node
+        Node* tmp = new Node();
+        tmp->data = element;
+        tmp->next = NULL;
+        tmp->prev = NULL;
+
+        if ( isEmpty() ) {
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: Braces should be on their own line.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        // Delete the front node and fix the links
+        Node* tmp = front;
+        if ( front->next != NULL )
+        {
+            front = front->next;
+            front->prev = NULL;
+        }
+        else
+        {
+            front = NULL;
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: For consistency, put also rear to nullptr.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        rear = NULL;
+        count = 0;
+    }
+
+    void InsertFront(int element)
+    {
+        // Create a new node
+        Node* tmp = new Node();
+        tmp->data = element;
+        tmp->next = NULL;
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: Use nullptr instead of NULL.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        // Delete the front node and fix the links
+        Node* tmp = rear;
+        if ( rear->prev != NULL )
+        {
+            rear = rear->prev;
+            rear->next = NULL;
+        }
+        else
+        {
+            rear = NULL;
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: Put also front to nullptr.

File: code/data_structures/src/queue/deque/deque.cpp [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+        // Pop elements
+        cout << q1.RemoveFront() << endl;
+        cout << q1.RemoveFront() << endl;
+        cout << q1.RemoveFront() << endl;
+    }
+    catch (...) {
+        cout << "Some exception occured" << endl;
+    }
+}
```
Comment (abdouskamel) [inline code/data_structures/src/queue/deque/deque.cpp]: Be sure to look at the C++ style guide.

**Comment by jainaman1398** [comment]
did it

**Comment by abdouskamel** [comment]
@jainaman1398 And for templates ?

**Pull request closed without merging**